third_party/move/move-model/bytecode/src/astifier.rs

```diff
@@ -1429,59 +1430,544 @@ struct IfElseTransformer<'a> {
 
 impl ExpRewriterFunctions for IfElseTransformer<'_> {
     fn rewrite_exp(&mut self, exp: Exp) -> Exp {
-        if let Some(result) = self.try_make_if(exp.clone()) {
-            self.rewrite_exp_descent(result)
-        } else {
-            self.rewrite_exp_descent(exp)
+        // Please maintain the order of the transformers.
+        let transformers: [fn(&Self, Exp) -> Option<Exp>; 4] = [
+            Self::try_remove_redundant_loop,
+            Self::try_make_if,
+            Self::try_make_if_else,
+            Self::try_remove_redundant_exp,
+        ];
+        let mut old_exp = exp;
+        // Recursively apply the transformers until a fixed point is reached.
+        loop {
```

Comment:
Do we want to put a limit on the number of times this loop is run? If not, could we briefly write down why we expect this loop to eventually reach a fixpoint (so that any new transformers added in the future know what they can/cannot do)?